test/test_ops.py

```diff
@@ -125,8 +125,31 @@ def test_inplace_gradgrad(self, device, dtype, op):
         self._gradgrad_test_helper(device, dtype, op, self._get_safe_inplace(op.get_inplace()))
 
 
+class TestOut(TestCase):
+    exact_dtype = True
+
+    @ops(op_db)
+    def test_out(self, device, dtype, op):
+        if not op.supports_out_param:
+            self.skipTest("Skipped! Operator %s does not support out=..." % op.name)
+        samples = op.sample_inputs(device, dtype)
+        if len(samples) == 0:
+            self.skipTest("Skipped! No sample inputs!")
+
+        # NOTE: only tests on first sample
+        sample = samples[0]
+        # call it normally to get the expected result
+        expected = op(sample.input, *sample.args, **sample.kwargs)
+        # call it with out=... and check we get the expected result
+        out_kwargs = sample.kwargs.copy()
+        out_kwargs['out'] = out = torch.empty(expected.shape, device=device, dtype=dtype)
```

Comment:
This won't work in general because not every op returns a tensor, and not every op that returns a tensor returns one on the same device and with the same dtype as the inputs. 

You could refine this test and the property you're setting on the OpInfos to be something like "supports_tensor_out" and then use `torch.empty_like(expected)` here. That will skip future OpInfos that take tuple outputs, but this test can be expanded to handle them properly later.

Reply:
fixed by 7d8b933ec2 and 3755d54667.

The alternative is to improve the test to automatically detect whether `expected` is a tuple and do "the right thing" to check that `out` matches, but OTOH we can't write this logic now because we don't have any suitable operator in the Op db